vowpalwabbit/learner.cc

```diff
@@ -237,12 +237,21 @@ class ready_examples_queue
 class custom_examples_queue
 {
 public:
-  custom_examples_queue(v_array<example*> examples) : _examples(examples) {}
+  void reset_examples(const v_array<example*>* examples)
+  {
+    assert(examples != nullptr);
+    _examples = examples;
+    _index = 0;
+  }
 
-  example* pop() { return _index < _examples.size() ? _examples[_index++] : nullptr; }
+  example* pop()
+  {
+    assert(_examples != nullptr);
+    return _index < _examples->size() ? (*_examples)[_index++] : nullptr;
+  }
 
 private:
-  v_array<example*> _examples;
+  const v_array<example*>* _examples;
```

Comment:
why was this made a pointer? who owns the `_examples` memory?

Reply:
A v_array is passed in it needs to be converted to this interface to be processed. In the past this relied on the shallow copy of the v_array to function efficiently. `_examples` is owned by whoever calls parse_dispatch and we are done with it by the end of the call